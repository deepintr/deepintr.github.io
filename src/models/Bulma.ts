export declare namespace Bulma {
  export type Size = 'small' | 'normal' | 'medium' | 'large';

  export type HeroSize = 'medium' | 'large' | 'fullheight';

  export type Color =
    | 'white'
    | 'black'
    | 'light'
    | 'dark'
    | 'primary'
    | 'link'
    | 'info'
    | 'success'
    | 'warning'
    | 'danger';

  export type Placement = 'left' | 'right';

  export type ColumnSize = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;
}
