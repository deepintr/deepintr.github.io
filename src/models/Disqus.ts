export declare namespace Disqus {
  export interface Config {
    shortname: string;
    config: {
      url: string;
      identifier: string;
      title: string;
    };
  }
}
