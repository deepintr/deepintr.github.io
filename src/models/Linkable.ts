export interface Linkable {
  href: string;
  isInternal?: boolean;
}
