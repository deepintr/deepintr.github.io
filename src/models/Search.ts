export interface SearchListing {
  title: string;
  slug: string;
}
