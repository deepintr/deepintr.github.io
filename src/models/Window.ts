interface Window {
  __FLEXSEARCH__: {
    [lang: string]: {
      index: {
        name: string;
        attrs?: {
          [key: string]: string;
        };
        values: any;
      }[];
      store: {
        id: number;
        node: {
          title: string;
          slug: string;
        };
      }[];
    };
  };
}
