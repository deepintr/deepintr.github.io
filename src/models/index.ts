export * from './Author';
export * from './Bulma';
export * from './Data';
export * from './FAIcon';
export * from './Linkable';
export * from './Menu';
