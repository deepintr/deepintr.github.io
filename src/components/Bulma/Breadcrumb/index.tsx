import React from 'react';
import { Link } from 'gatsby';
import clsx from 'clsx';
import { Menu } from '../../../models';

export interface BreadcrumbProps {
  items: Menu.NavLink[];
}

const Breadcrumb: React.FC<BreadcrumbProps> = ({ items }) => {
  const parts: Menu.NavLink[] = [
    {
      name: 'Anasayfa',
      url: {
        href: '/',
      },
    },
    ...items,
  ];

  return (
    <nav className="breadcrumb has-arrow-separator" aria-label="breadcrumbs">
      <ul>
        {parts.map(({ name, url }, i) => (
          <li
            key={`breadcrumb-item-${i}`}
            className={clsx({ 'is-active': i === parts.length - 1 })}
          >
            {/* Treat the all links as internal. */}
            <Link to={url.href}>{name}</Link>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Breadcrumb;
