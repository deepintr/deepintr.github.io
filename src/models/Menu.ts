import { FAIcon } from './FAIcon';
import { Linkable } from './Linkable';

export declare namespace Menu {
  export interface IMenuItem {
    [key: string]: any;
    name: string;
    icon: FAIcon.Icon;
  }

  export interface ItemRegular extends IMenuItem {
    url: Linkable;
  }

  export interface ItemDropdown extends IMenuItem {
    items?: ItemRegular[];
  }

  export type ItemType = ItemRegular | ItemDropdown;

  export interface ContactItem extends ItemRegular {
    isFeatured?: boolean;
  }

  export interface NavLink {
    name: string;
    url: Linkable;
  }

  export interface DownloadLink {
    fileName: string;
    sourceName: string;
    url: string;
  }
}
