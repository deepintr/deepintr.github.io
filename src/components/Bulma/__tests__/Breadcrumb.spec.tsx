import React from 'react';
import renderer from 'react-test-renderer';
import Breadcrumb from '../Breadcrumb';
import { Menu } from '../../../models';

describe('<Breadcrumb />', () => {
  test('should render correctly', () => {
    const items: Menu.NavLink[] = [
      {
        name: 'Test',
        url: {
          href: '/test',
          isInternal: true,
        },
      },
    ];
    const tree = renderer.create(<Breadcrumb items={items} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
