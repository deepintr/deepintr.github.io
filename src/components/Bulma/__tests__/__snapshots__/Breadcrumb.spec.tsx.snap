// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb /> should render correctly 1`] = `
<nav
  aria-label="breadcrumbs"
  className="breadcrumb has-arrow-separator"
>
  <ul>
    <li
      className=""
    >
      <a
        href="/"
      >
        Anasayfa
      </a>
    </li>
    <li
      className="is-active"
    >
      <a
        href="/test"
      >
        Test
      </a>
    </li>
  </ul>
</nav>
`;
